File: 2032-two-out-of-three/2032-two-out-of-three.java
class Solution {
    public List<Integer> twoOutOfThree(int[] nums1, int[] nums2, int[] nums3) {
        HashSet<Integer> result= new HashSet<>();
        HashSet<Integer> set1= new HashSet<>();
        HashSet<Integer> set2= new HashSet<>();
        
        for(int i=0; i<nums1.length; i++)
        {
            set1.add(nums1[i]);
        }
        
        for(int i=0; i<nums2.length;i++)
        {
            if(set1.contains(nums2[i]))
            {
                result.add(nums2[i]);
            }
            else
            {
                set2.add(nums2[i]);
            }
        }
        
        for(int i=0; i<nums3.length;i++)
        {
            if(set1.contains(nums3[i]) || set2.contains(nums3[i]))
                result.add(nums3[i]);
        }
        return new ArrayList(result);
    }
    
}

File: 2032-two-out-of-three/2032-two-out-of-three.cpp
class Solution {
public:
    vector<int> twoOutOfThree(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3) {
        unordered_map<int, set<int>> mp;
        vector<int> result;
        
        for(int i: nums1)
        {
            mp[i].insert(0);
        }
        for(int i: nums2)
        {
            mp[i].insert(1);
        }
        for(int i: nums3)
        {
            mp[i].insert(2);
        }
        
        for(auto it: mp)
        {
            if(it.second.size()>=2)
            {
                result.push_back(it.first);
            }
        }
        return result;
    }
};